### DISCUSSION TITLE
Apply method of a module

### DISCUSSION THREAD

**Discussion by goktugguvercin** [google/flax#3298] [Q&A] [answered]
Hello Flax community,

I have a resnet architecture with batch-norm layers. While doing forward pass by `resnet.apply()`, I am confused at some point:
[Documentation](https://flax.readthedocs.io/en/latest/guides/batch_norm.html) specifies that the argument `batch_stats` needs to be marked as mutable. **Is this valid for every case or just `train=True` ?**

For example, I am in testing phase, and I want to check performance of my model with specific metrics. In that case, I will use my test data, and batch norm layers will use its statistics. **So, batch_stats will not be updated right; I think no need for mutable argument in testing phase ? What do you think ?**

Thanks in advance.

**Comment by cgarciae** [comment, marked as answer]
Hey @GoktugGuvercin, you only need to specify `mutable=True` when `use_running_average=False`. If you are using the usual `train` argument, you don't need to specify `mutable` when `train=False` i.e. during inference.

**Reply by goktugguvercin** [reply]
Ah okey. Thank you so much for the clarification :)
